File: lib_video/src/main/java/com/effective/android/video/interfaces/IControlLayer.java
package com.effective.android.video.interfaces;

import com.effective.android.video.bean.VideoCache;

/**
 * 视图层通过该接口控制播放器行为
 * created by yummylau 2019/04/20
 */
public interface IControlLayer {

    void restore(VideoCache videoCache);

    void replay(VideoCache videoCache);

    void play(VideoCache videoCache);

    void pause(VideoCache videoCache,boolean userAction);

    void stop(VideoCache videoCache);

    void release(VideoCache videoCache);

    void seekTo(VideoCache videoCache, long position);
}

File: lib_video/src/main/java/com/effective/android/video/core/IControlLayer.kt
package com.effective.android.video.core

import com.effective.android.video.bean.VideoCache

/**
 * 视图层通过该接口控制播放器行为
 * created by yummylau 2019/04/20
 */
interface IControlLayer {

    fun restore(videoCache: VideoCache)

    fun replay(videoCache: VideoCache)

    fun play(videoCache: VideoCache)

    fun pause(videoCache: VideoCache, userAction: Boolean)

    fun stop(videoCache: VideoCache)

    fun release(videoCache: VideoCache)

    fun seekTo(videoCache: VideoCache, position: Long)
}
